### PR TITLE
Update drop down

### PR DESCRIPTION
**Pull request by CodingWithCDJE**
* Cleaner code in app & dropdown component
* Alphabetize functionally works

### CODE REVIEW

File: src/components/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -1,31 +1,86 @@
...
-    constructor(props){
-        super(props);
-        this.state = {
-            data : this.props.data,
-            options : ["Select Category", "Alphabetical", "Project Start Date", "Project End Date", "Satisfaction","Budget", "Number of Blockers", "Engineering Manager", "Project Manager"]
-        }
-    }
+  alphabetize() {
+    const cloneData = this.state.theData;
+    const sortedAlphabeticalData = cloneData.sort((a, b) => {
```
Comment (horseeyephil) [inline src/components/Dropdown/Dropdown.js]: 🙅 - is this really cloning the data??

Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: No, that wasn't it assigned the pointer of where the state data is to the variable cloneData. I fixed this by using the slice method.

Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: No it is not. What that did was assigning the pointer of where the state data to the variable cloneData. I fixed this by using the slice method.

File: server/list-view.json [hunk 2]
```diff
@@ -34,8 +34,8 @@
                     "organization": ""
                 }
             ],
-            "startDate": "01-29-20",
-            "endDate": "04-20-20",
+            "startDate": "01-29-2020",
```
Comment (richsong) [inline server/list-view.json]: These updates need to be a separate pull request..

File: src/components/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -1,31 +1,77 @@
...
     }
-}
+    if (a.name > b.name) {
+      return 1;
+    }
+    return 0;
+  });
+  props.updateDataState(sortedAlphabeticalData);
+};
+
```
Comment (horseeyephil) [inline src/components/Dropdown/Dropdown.js]: 👍 - great job!

File: src/App.js [hunk 4]
```diff
@@ -1,55 +1,51 @@
...
-  }, 1000)     
-}
+  sortDataState(sortData) {
+    this.setState({ theData: sortData });
+  }
 
+  render() {
+    const { theData } = this.state;
 
+    const options = [
```
Comment (richsong) [inline src/App.js]: I thought we agreed that options would be coming from the JSON payload?

File: src/App.js [hunk 5]
```diff
@@ -1,55 +1,51 @@
...
-    });
-    console.log(this.state.theData)
-  }, 1000)     
-}
+  sortDataState(sortData) {
+    this.setState({ theData: sortData });
+  }
 
+  render() {
+    const { theData } = this.state;
```
Comment (richsong) [inline src/App.js]: Please rename this to something other than `theData`

File: src/components/Dropdown/Dropdown.js [hunk 6]
```diff
@@ -1,31 +1,77 @@
...
 
-class Dropdown extends Component {
-    constructor(props){
-        super(props);
-        this.state = {
-            data : this.props.data,
-            options : ["Select Category", "Alphabetical", "Project Start Date", "Project End Date", "Satisfaction","Budget", "Number of Blockers", "Engineering Manager", "Project Manager"]
-        }
-    }
+const Sortby = (event, props) => {
```
Comment (horseeyephil) [inline src/components/Dropdown/Dropdown.js]: let's keep this lowercase - remember, typical programming convention is to leave Uppercase for class singletons or types
-exception**: also in React, we use them for React components (even if they are just functions)

Comment (richsong) [inline src/components/Dropdown/Dropdown.js]: change to `SortBy`

File: src/components/Dropdown/Dropdown.js [hunk 7]
```diff
@@ -1,31 +1,77 @@
...
+      </select>
+    </div>
+  );
+};
 
+const RenderOptions = props => {
+  return props.map(i => {
+    return (
+      <option key={i} value={i}>
+        {i}
```
Comment (richsong) [inline src/components/Dropdown/Dropdown.js]: What is `i` here? Please rename argument to something more verbose

File: src/components/Dropdown/Dropdown.js [hunk 8]
```diff
@@ -1,31 +1,77 @@
...
-            data : this.props.data,
-            options : ["Select Category", "Alphabetical", "Project Start Date", "Project End Date", "Satisfaction","Budget", "Number of Blockers", "Engineering Manager", "Project Manager"]
-        }
-    }
+const Sortby = (event, props) => {
+  switch (event) {
+    case "Alphabetical":
+      SortByAlphabetical(props);
+
+    case "Project Start Date":
```
Comment (richsong) [inline src/components/Dropdown/Dropdown.js]: Where is the rest of this functionality? You need a sort function for each of these cases.

File: src/App.js [hunk 9]
```diff
@@ -1,55 +1,51 @@
...
-  
+class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      theData: [],
-      isLoading: false,  
-   }
-    this.handleClick = this.handleClick.bind(this);
+      theData: data.projects
```
Comment (horseeyephil) [inline src/App.js]: now that the state of this master is looking much better! let's rename `theData` to something more semantically useful~
why not `projects` or `displayProjects`

File: src/components/Dropdown/Dropdown.js [hunk 10]
```diff
@@ -1,31 +1,77 @@
...
+      break;
+
+    default:
+      props.updateDataState(props.originalData);
+      break;
+  }
+};
+
+const SortByAlphabetical = props => {
+  const cloneData = props.theData.slice();
```
Comment (richsong) [inline src/components/Dropdown/Dropdown.js]: What's the point of this slice? Just assign props.data to a given variable.

File: src/App.js [hunk 11]
```diff
@@ -1,55 +1,51 @@
...
-    this.setState({
-      theData: data.projects,
-      isLoading: false
-    });
-    console.log(this.state.theData)
-  }, 1000)     
-}
+  sortDataState(sortData) {
+    this.setState({ theData: sortData });
+  }
```
Comment (horseeyephil) [inline src/App.js]: totally extra: but for contained cases like this,  I enjoy this kind of syntax 😝 
` sortDataState(theData) {
  this.setState({ theData })
}`

File: src/components/card/ProjectListItem.js [hunk 12]
```diff
@@ -1,60 +1,64 @@
 import React, { Component } from "react";
-import Bag from '../../../svg/bag.svg';
-import Calendar from '../../../svg/calendar.svg';
-import Clock from '../../../svg/clock.svg';
-import Arrow from '../../../svg/arrow.svg';
+import Bag from "../../../svg/bag.svg";
```
Comment (richsong) [inline src/components/card/ProjectListItem.js]: Changes for ProjectListItem should be in separate PR

File: src/App.js (hunk 11 above)
Comment (horseeyephil) [reply src/App.js]: but be careful - because it can lead to runtime bugs if you make a typo

File: server/list-view.json (hunk 2 above)
Comment (CodingWithCDJE) [reply server/list-view.json]: Got it for the next time unless you want me to revert changes.

File: src/components/Dropdown/Dropdown.js [hunk 13]
```diff
@@ -1,31 +1,77 @@
...
+    case "Project Manager":
+      break;
+
+    default:
+      props.updateDataState(props.originalData);
+      break;
+  }
+};
+
+const SortByAlphabetical = props => {
```
Comment (horseeyephil) [inline src/components/Dropdown/Dropdown.js]: same here

File: src/components/Dropdown/Dropdown.js (hunk 10 above)
Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: The point for slice is to get a clone of the  project data, because if we were to assign props.data then it doesn't change back to the default values since the data remains to be sorted, even though what I am passing in is the JSON payload.

Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: So by doing slice, it doesn't affect the payload, but just affects the state.

File: src/App.js (hunk 4 above)
Comment (CodingWithCDJE) [reply src/App.js]: Did the requested change on another branch.

File: src/App.js (hunk 5 above)
Comment (CodingWithCDJE) [reply src/App.js]: renamed it to projectData

File: src/components/Dropdown/Dropdown.js (hunk 7 above)
Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: I renamed it to be optionName.

File: src/App.js (hunk 9 above)
Comment (CodingWithCDJE) [reply src/App.js]: changed the state to be projectData

File: src/components/card/ProjectListItem.js (hunk 12 above)
Comment (CodingWithCDJE) [reply src/components/card/ProjectListItem.js]: I apologize for this.

File: src/components/Dropdown/Dropdown.js (hunk 13 above)
Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: corrected my mistake

File: src/components/Dropdown/Dropdown.js (hunk 8 above)
Comment (CodingWithCDJE) [reply src/components/Dropdown/Dropdown.js]: Yes. I have only done the alphabetical one, but I am moving on to the next one.
